fem/nonlininteg.hpp

```diff
@@ -72,6 +72,42 @@ class NonlinearFormIntegrator
 };
 
 
+class BlockNonlinearFormIntegrator
+{
+public:
+   /// Compute the local energy
+   virtual double GetElementEnergy(Array<const FiniteElement *>&el,
+                                   ElementTransformation &Tr,
+                                   Array<const Vector *>&elfun);
```

Comment:
It may be a bit more expressive to switch the `Array<...>` types to `const Array<...>`. What do you think?

Reply:
That makes sense to me. I'll try it out and see if any issues pop up.